DICOM/src/Microsoft.Health.DeID.SharedLib/EncryptFunction.cs

```diff
@@ -7,16 +7,29 @@
 using System.IO;
 using System.Security.Cryptography;
 using System.Text;
-using Newtonsoft.Json;
+using Microsoft.Health.DeID.SharedLib.Settings;
 
 namespace Microsoft.Health.Dicom.DeID.SharedLib
 {
     public class EncryptFunction
     {
         // AES Initialization Vector length is 16 bytes
         private const int AesIvSize = 16;
+        private readonly byte[] _aesKey;
+        private readonly byte[] _privateKey;
+        private readonly byte[] _publicKey;
 
-        public static byte[] EncryptContentWithAES(string plainText, byte[] key, Encoding encoding = null)
+        public EncryptFunction(EncryptionSetting encryptionSetting = null)
+        {
+            encryptionSetting ??= new EncryptionSetting();
+            encryptionSetting.Validate();
+
+            _aesKey = Encoding.UTF8.GetBytes(encryptionSetting.EncryptKey ?? Guid.NewGuid().ToString("N"));
```

Comment:
>encryptionSetting.EncryptKey ?? Guid.NewGuid().ToString("N") [](http://example.com/codeflow?start=45&length=60)

this has something to rethink about. Do you know how many bits encryption is used in the setting. You are using a guid which limits the key length to 256 bit, but I am not sure if we should use this kind of encryption or maybe more bits.

Reply:
Acceptable sizes are 128, 192, 256. 256 is applicable here.